tasklistapiOAS.yaml

```diff
@@ -0,0 +1,130 @@
+openapi: 3.0.3
+info:
+  title: Task List App API
+  version: 1.0.0
+  description: API for managing tasks in a task list application
+  contact:
+    name: Jackson Geer
+    email: jacksongeer@gmail.com
+    url: https://github.com/skeptickal/TaskListBackend
+servers:
+  - url: "http://localhost:8080"
+paths:
+  /tasks:
+    get:
+      tags:
+        - Task List App Functions
+      summary: Get all tasks
+      parameters:
+        - name: status
+          in: query
+          description: Filter tasks by status
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Task'
+        '400':
+          description: Bad request
+    post:
+      tags:
+          - Task List App Functions
+      summary: Create a new task
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Task'
+      responses:
+        '201':
+          description: Task created successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Task'
+        '400':
+          description: Bad request
+  /tasks/{id}:
+    put:
+      tags:
+          - Task List App Functions
+      summary: Update a task
+      parameters:
+        - name: id
+          in: path
+          required: true
+          schema:
+            type: integer
+            format: int64
+        - name: newStatus
```

Comment:
I don't think this should exist because you can already set the new status from the response body.

Reply:
got it